apps/isitchristmas/is_it_christmas.star

```diff
@@ -105,8 +105,6 @@ def main(config):
 def get_daysleft(today, timezone, christmas_month, christmas_day):
     if today.month == christmas_month and today.day > christmas_day:
         year = today.year + 1
-    if today.month > christmas_month:
```

Comment:
Not sure this makes sense.

Let's say Christmas is 6 Jan, and today is February 2023. In that case, you need to add another year because we'll be counting down 11 months to 6 Jan 2024.

If Christmas is December, then this can't be reached anyway because it can't be a greater month than December.

Reply:
You are correct, I only checked christmas in December.... I don't think both need to be there, or it needs to be a elif. Let me fix that. 

That's what I get for not doing proper edge checking...

Think I fixed it with the latest commit, the today.month needed >= in order to roll over correctly. Since any time the date has passed a year needs to be added. 

```
    if today.month >= christmas_month and today.day > christmas_day:
        year = today.year + 1
    else:
        year = today.year
```

Reply:
I see the issue you're fixing, but I think it needs to be something like this, right?

```
    if today.month == christmas_month and today.day > christmas_day:
        year = today.year + 1  # During month of Christmas, but after Christmas has passed, so look to next year
    elif today.month > christmas_month:
        year = today.year + 1  # Christmas is over this year, so count to next year
    else:
        year = today.year  # Christmas is still to come this year
```

Suppose you use `today.month >= christmas_month and today.day > christmas_day` and Christmas is 6 January. Then on 2 February 2023, you'd end up counting down negative days to 6 Jan 2023, instead of forward to 6 Jan 2024.

I did it this way because the day of the month only matters during the month of Christmas. After that, you don't care about days, you can just do it all with months.

But you're right that it should have been an elif.

Reply:
Darn it, yeah, you're right. I was wondering when that elif edge case was gonna need to be present. 

Originally, a year was getting added twice anyways I think because it was merely if and then another if else statement, which was wrong in and of itself. 

The actual error originally then, which I thought was going to be the original fix was changing that to elif, and then I thought I could eliminate that case... 

Anyways I've implemented that simple change back in. Thanks for the sanity check!